src/corelibs/U2Gui/src/util/project/ProjectTreeController.cpp

```diff
@@ -792,8 +795,64 @@ void ProjectTreeController::sl_onRemoveSelectedItems() {
     const QList<Document *> selectedDocs = getDocsInSelection(deriveDocsFromObjs).values();
     const QList<Folder> selectedFolders = getSelectedFolders();
     const QList<GObject *> selectedObjects = objectSelection.getSelectedObjects();
+    
+    QStringList selectedItemNames;
+    foreach(Document * doc, selectedDocs) {
+        selectedItemNames.append(doc->getName());
+    }
+    foreach(const Folder& folder, selectedFolders) {
+        selectedItemNames.append(folder.getFolderName());
+    }
+    foreach(GObject* obj, selectedObjects) {
+        selectedItemNames.append(obj->getGObjectName());
+    }
+    
+    QString fullItemsNamesList;
+
+    QString warningMessageText = (1 == selectedItemNames.count()) ? tr("Remove selected item?") : tr("Remove selected items?");
```

Comment:
The warning message is incorrect. Check the issue details for the correct message.